File: client/httpclient.js
const rp = require('request-promise');
const url = 'http://127.0.0.1:5555';
const HEART_BEAT = 5005;
const RECV_TXT_MSG = 1;
const RECV_PIC_MSG = 3;
const USER_LIST = 5000;
const GET_USER_LIST_SUCCSESS = 5001;
const GET_USER_LIST_FAIL     = 5002;
const TXT_MSG = 555;
const PIC_MSG = 500;
const AT_MSG = 550;
const CHATROOM_MEMBER = 5010;
const CHATROOM_MEMBER_NICK = 5020;
const PERSONAL_INFO = 6500;
const DEBUG_SWITCH = 6000;
const PERSONAL_DETAIL =6550;
function getid()
{
   const id = Date.now();
   return id.toString();
}
async function send_at_msg()
{
    const options =
      {
          //method: 'GET',
          //url: 'https://apis.map.qq.com/ws/district/v1/list',
          url: url+'/api/sendatmsg',
          qs:
          {
              id: getid(),
              type: AT_MSG,
              roomid: '23023281066@chatroom',//the chatroom 
              wxid:'the wxid who you  want to send',
              content: '今天过来一起喝酒！同时，我们也发现',
              nick:'[微笑]Python'
          }
      };
      let data = await rp(options);
      const j = JSON.parse(data);
      
      //console.log(j.id); 
      //console.log(j.status);
      return j;


}
/**send_pic
 *  发送图片
 */
async function send_pic()
{
    const options =
      {
          //method: 'GET',
          //url: 'https://apis.map.qq.com/ws/district/v1/list',
          url: url+'/api/sendpic',
          qs:
          {
              id: getid(),
              type: PIC_MSG,
              wxid: 'filehelper',
              path: 'C:\\Users\\14988\\Desktop\\temp\\2.jpg'
          }
      };
      let data = await rp(options);
      const j = JSON.parse(data);
      
      //console.log(j.id); 
      //console.log(j.status);
      return j;


}

/**get_member_nick
 * 获取群成员昵称
 */

async function get_member_nick()
{
    //getmembernick
    const options =
      {
          
          url: url+'/api/getmembernick',
          qs:
          {
              id: getid(),
              type: CHATROOM_MEMBER_NICK,
              roomid: '17339716569@chatroom'//1280641435@chatroom
          }
      };
    const data = await rp(options);
    const j = JSON.parse(data);
    
    //console.log(j.id); 
    //console.log(j.status);
    return j;
}
/**get_getmemberid
 *   获取群成员id
 */
async function get_memberid()
{
    const options={
        url: url+'/api/getmemberid'
    };
    let data = await rp(options);
    const j = JSON.parse(data);
    
    //console.log(j.id); 
    //console.log(j.status);
    return j;
}

/**get_contact_list  获取好友和群列表
 * 
 */
async function get_contact_list()
{
    const options={
        url: url+'/api/getcontactlist'
    };
    let data = await rp(options);
    const j = JSON.parse(data);
    
    //console.log(j.id); 
    //console.log(j.status);
    return j;

}
/**
 * send txt message
 */
async function sendtxt()
{
    const options =
      {
          //method: 'GET',
          //url: 'https://apis.map.qq.com/ws/district/v1/list',
          url: url+'/api/sendtxtmsg',
          qs:
          {
              id: getid(),
              type: TXT_MSG,
              wxid: 'filehelper',
              content: '今天我寒夜里看学飘过2338？？&&2222！'
          }
      };
      let data = await rp(options);
      const j = JSON.parse(data);
      
      //console.log(j.id); 
      //console.log(j.status);
      return j;


}
/**send_attatch
 * send the attatchment
 */
async function send_attatch()
{
    const options =
      {
          //method: 'GET',
          //url: 'https://apis.map.qq.com/ws/district/v1/list',
          url: url+'/api/sendattatch',
          qs:
          {
              id: getid(),
              //type: TXT_MSG,
              wxid: '23023281066@chatroom',
              filepath: 'C:\\temp\\wechat.txt'
          }
      };
      let data = await rp(options);
      const j = JSON.parse(data);
      
      //console.log(j.id); 
      //console.log(j.status);
      return j;
}

async function refresh_memberlist()
{
    const options =
      {
          //method: 'GET',
          //url: 'https://apis.map.qq.com/ws/district/v1/list',
          url: url+'/api/refresh_chatroom',
          qs:
          {
              id: getid(),
              wxid: 'null',
              content: 'null'
          }
      };
      let data = await rp(options);
      const j = JSON.parse(data);
      
      //console.log(j.id); 
      //console.log(j.status);
      return j;
}
async function send_destroy()
{
    const options =
      {
          //method: 'GET',
          //url: 'https://apis.map.qq.com/ws/district/v1/list',
          url: url+'/api/destroy',
          qs:
          {
              id: getid(),
              wxid: 'null',
              content: 'null'
          }
      };
      let data = await rp(options);
      const j = JSON.parse(data);
      
      //console.log(j.id); 
      //console.log(j.status);
      return j;
}

async function main()
{
    //const j = await get_contact_list();
    //const j = await get_memberid();
    //const j = await get_member_nick();
    //const j = await send_at_msg();
    //const j = await send_attatch();
    //console.log(j);
    //await send_destroy();
    await refresh_memberlist();
}

//console.log('test');
main();

File: client/python/http-client.py
# pip install requests

import requests
import json
from datetime import datetime

url = 'http://127.0.0.1:5555'
HEART_BEAT = 5005
RECV_TXT_MSG = 1
RECV_PIC_MSG = 3
USER_LIST = 5000
GET_USER_LIST_SUCCESS = 5001
GET_USER_LIST_FAIL = 5002
TXT_MSG = 555
PIC_MSG = 500
AT_MSG = 550
CHATROOM_MEMBER = 5010
CHATROOM_MEMBER_NICK = 5020
PERSONAL_INFO = 6500
DEBUG_SWITCH = 6000
PERSONAL_DETAIL = 6550
ATTATCH_FILE = 5003

def get_id():
    return str(int(datetime.now().timestamp() * 1000))

def send_at_msg():
    jpara = {
        'id': get_id(),
        'type': AT_MSG,
        'roomid': '23023281066@chatroom',
        'wxid': 'zhanghua_cd',
        'content': 'at msg test,hello world，真的有一套',
        'nickname': '老张',
        'ext': 'null'
    }
    response = requests.post(url + '/api/sendatmsg', json={'para': jpara})
    return response.json()

def send_pic():
    jpara = {
        'id': get_id(),
        'type': PIC_MSG,
        'wxid': '23023281066@chatroom',
        'roomid': 'null',
        'content': 'C:\\tmp\\2.jpg',
        'nickname': "null",
        'ext': 'null'
    }
    response = requests.post(url + '/api/sendpic', json={'para': jpara})
    return response.json()

def get_member_nick(wx_id, roomid):
    jpara = {
        'id': get_id(),
        'type': CHATROOM_MEMBER_NICK,
        'wxid': 'zhanghua_cd',
        'roomid': '23023281066@chatroom',
        'content': 'null',
        'nickname': "null",
        'ext': 'null'
    }
    response = requests.post(url + '/api/getmembernick', json={'para': jpara})
    return response.json()

def get_memberid():
    jpara = {
        'id': get_id(),
        'type': CHATROOM_MEMBER,
        'wxid': 'null',
        'content': 'op:list member'
    }
    response = requests.post(url + '/api/getmemberid', json={'para': jpara})
    return response.json()

def get_contact_list():
    jpara = {
        'id': get_id(),
        'type': USER_LIST,
        'roomid': 'null',
        'wxid': 'null',
        'content': 'null',
        'nickname': 'null',
        'ext': 'null'
    }
    response = requests.post(url + '/api/getcontactlist', json={'para': jpara})
    return response.json()

def get_chatroom_member_list():
    jpara = {
        'id': get_id(),
        'type': CHATROOM_MEMBER,
        'roomid': 'null',
        'wxid': 'null',
        'content': 'null',
        'nickname': 'null',
        'ext': 'null'
    }
    response = requests.post(url + '/api/get_charroom_member_list', json={'para': jpara})
    return response.json()

def send_txt_msg():
    jpara = {
        'id': get_id(),
        'type': TXT_MSG,
        'wxid': '23023281066@chatroom',
        'roomid': 'null',
        'content': 'hello word',
        'nickname': "null",
        'ext': 'null'
    }
    response = requests.post(url + '/api/sendtxtmsg', json={'para': jpara})
    return response.json()

def send_attatch():
    jpara = {
        'id': get_id(),
        'type': ATTATCH_FILE,
        'wxid': '23023281066@chatroom',
        'roomid': 'null',
        'content': 'C:\\tmp\\log.7z',
        'nickname': "null",
        'ext': 'null'
    }
    response = requests.post(url + '/api/sendattatch', json={'para': jpara})
    return response.json()

def main():
    response = get_contact_list()
    # response = get_chatroom_member_list()
    # response = send_txt_msg()
    # response = get_member_nick("zhanghua_cd", "23023281066@chatroom")
    # response = send_at_msg()
    # response = send_attatch()
    # response = get_member_nick()
    # print(response)
    # await send_destroy()
    # await refresh_memberlist()
    # print("test begin")
    # response = send_destroy()
    # response = send_attatch()
    print(json.dumps(response, indent=2))

if __name__ == '__main__':
    main()
